keras_retinanet/preprocessing/open_images.py

```diff
@@ -0,0 +1,206 @@
+import csv
+import json
+import os
+
+import numpy as np
+from PIL import Image
+
+from .generator import Generator
+from ..utils.image import read_image_bgr
+
+
+def get_labels(metadata_dir):
+    trainable_classes_path = os.path.join(metadata_dir, 'classes-bbox-trainable.txt')
+    description_path = os.path.join(metadata_dir, 'class-descriptions.csv')
+
+    description_table = {}
+    with open(description_path) as f:
+        for row in csv.reader(f):
+            if len(row):
+                description_table[row[0]] = row[1].replace("\"", "").replace("'", "").replace('`', '')
+
+    with open(trainable_classes_path, 'rb') as f:
+        trainable_classes = f.read().split('\n')
+
+    id2labels_dict = dict([(i, description_table[c]) for i, c in enumerate(trainable_classes)])
+    cls_index_dict = dict([(c, i) for i, c in enumerate(trainable_classes)])
+
+    return id2labels_dict, cls_index_dict
+
+
+def generate_images_annotations_json(main_dir, metadata_dir, subset, cls_index_dict):
+    annotations_path = os.path.join(metadata_dir, subset, 'annotations-human-bbox.csv')
+
+    cnt = 0
+    with open(annotations_path, 'r') as csvfile:
+        reader = csv.DictReader(csvfile,
+                                fieldnames=['ImageID', 'Source', 'LabelName',
+                                            'Confidence', 'XMin', 'XMax', 'YMin',
+                                            'YMax'])
+        reader.next()
+        for _ in reader:
+            cnt += 1
+
+        print ('There are {} lines in subset: {}'.format(cnt, subset))
+
+    id_annotations = dict()
+
+    with open(annotations_path, 'r') as csvfile:
+        reader = csv.DictReader(csvfile,
+                                fieldnames=['ImageID', 'Source', 'LabelName',
+                                            'Confidence', 'XMin', 'XMax', 'YMin',
+                                            'YMax'])
+        reader.next()
+
+        images_sizes = dict()
+        for line, row in enumerate(reader):
+            frame = row['ImageID']
+
+            class_name = row['LabelName']
+
+            if class_name not in cls_index_dict:
+                continue
+
+            cls_id = cls_index_dict[class_name]
+
+            img_path = os.path.join(main_dir, 'images', subset, frame + '.jpg')
```

Comment:
I assume all images in the OID are `jpg` ? In case someone wants to use the OID format, but use `png` files, do you think it would be nice to have the extension configurable (but defaulting to `jpg`) ?

Reply:
Yes. They are all JPGs. 
It would be better to have a list of extension to scan eventually, but i prefer to have more feedback before doing that.